pkg/agent/agent_windows.go

```diff
@@ -0,0 +1,95 @@
+// +build windows
+
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agent
+
+import (
+	"strings"
+
+	"github.com/Microsoft/hcsshim"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/config"
+	"github.com/vmware-tanzu/antrea/pkg/agent/interfacestore"
+	"github.com/vmware-tanzu/antrea/pkg/agent/util"
+)
+
+// setupExternalConnectivity installs OpenFlow entries to SNAT Pod traffic using Node IP, and then Pod could communicate
+// to the external IP address.
+func (i *Initializer) setupExternalConnectivity() error {
+	subnetCIDR := i.nodeConfig.PodCIDR
+	nodeIP := i.nodeConfig.NodeIPAddr.IP
+	// Install OpenFlow entries on the OVS to enable Pod traffic to communicate to external IP addresses.
+	if err := i.ofClient.InstallExternalFlows(nodeIP, *subnetCIDR); err != nil {
+		klog.Errorf("Failed to setup SNAT openflow entries: %v", err)
+		return err
+	}
+	return nil
+}
+
+// prepareOVSBridge adds local port and uplink to ovs bridge.
+func (i *Initializer) prepareOVSBridge() error {
+	hnsNetwork, err := hcsshim.GetHNSNetworkByName(util.LocalHNSNetwork)
+	if err != nil {
+		return err
+	}
+	uplink := hnsNetwork.NetworkAdapterName
+
+	// Set datapathID of OVS bridge.
+	// If no datapathID configured explicitly, the reconfiguration operation will change OVS bridge datapathID
+	// and break the OpenFlow channel.
+	datapathID := strings.Replace(hnsNetwork.SourceMac, ":", "", -1)
+	datapathID = "0000" + datapathID
+	if err = i.ovsBridgeClient.SetDatapathID(datapathID); err != nil {
+		klog.Errorf("Failed to set datapath_id %s: %v", datapathID, err)
+		return err
+	}
+
+	// Create local port.
+	brName := i.ovsBridgeClient.GetBridgeName()
+	if _, err := i.ovsBridgeClient.GetOFPort(brName); err == nil {
+		klog.Infof("OVS bridge local port %s already exists, skip the configuration", brName)
+	} else {
+		if _, err := i.ovsBridgeClient.CreateInternalPort(brName, config.AutoAssignedOFPort, nil); err != nil {
+			return err
+		}
+	}
+
+	// If uplink is already exists, return.
+	if _, err := i.ovsBridgeClient.GetOFPort(uplink); err == nil {
+		klog.Infof("Uplink %s already exists, skip the configuration", uplink)
+		return err
+	}
+	// Create uplink port.
+	uplinkPortUUId, err := i.ovsBridgeClient.CreateUplinkPort(uplink, uplink, config.UplinkOFPort, nil)
```

Comment:
If uplink port and interface name are always the same, you might consider to save one parameter for CreateUplinkPort().

Reply:
Thanks Jianjun. Removed the "ifDev" param.